src/editor/esphome-editor.ts

```diff
@@ -0,0 +1,289 @@
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
+import { LitElement, html, css } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import { createRef, Ref, ref } from "lit/directives/ref.js";
+// @ts-ignore
+import styles from "monaco-editor/min/vs/editor/editor.main.css";
+import { fireEvent } from "../util/fire-event";
+
+// WebSocket URL Helper
+const loc = window.location;
+const wsLoc = new URL("./", `${loc.protocol}//${loc.host}${loc.pathname}`);
+wsLoc.protocol = "ws:";
+if (loc.protocol === "https:") {
+  wsLoc.protocol = "wss:";
+}
+const wsUrl = wsLoc.href;
+
+let editorActiveFilename: string | null = null;
+let editorActiveSecrets = false;
+let editorActiveWebSocket: WebSocket | null = null;
+let editorValidationScheduled = false;
+let editorValidationRunning = false;
+
+@customElement("esphome-editor")
```

Comment:
Have you considered using https://github.com/vanillawc/wc-monaco-editor

Reply:
no, in fact I tried other alternatives without much success. This project you point is a bit unmaintained and also I could not understand how to get access to the monaco api with this (may be I just don't get it) 
Do you really want to focus on this library or is this just a shallow question?
also I'm not sure why would we use that one instead of what I've done now

Reply:
The main reason would be maintenance. I didn't look into that web component specifically. More a generic question why not re-use something that will figure out all the CSS etc for us.